Revert removing Table of Contents from Code Reference and fix outside content instead

adamwoodnz opened the pull request:
See #76 

We initially tried removing the Table of Contents entirely on Code Reference pages, however [this proved controversial](https://make.wordpress.org/meta/2022/07/01/exploration-improving-devhub/#comment-9329), so I want to put this solution out for comment.

This PR makes the ToC fixed to the right of the content on wide screens for all handbooks. On medium and small screens it's displayed inline below the title.

Code Reference wide
![Screen Shot 2022-07-14 at 3 43 13 PM](https://user-images.githubusercontent.com/1017872/178893450-96217192-84fc-434e-a280-f60f19a5116d.jpg)

Code reference wide scrolled
![Screen Shot 2022-07-14 at 3 44 14 PM](https://user-images.githubusercontent.com/1017872/178893537-f3d3f400-92a3-40d3-b95d-aa0d19fef7b9.jpg)

Code Reference iPad
![Screen Shot 2022-07-14 at 3 44 58 PM](https://user-images.githubusercontent.com/1017872/178893632-09312fb9-2161-4ff7-ab62-81202530b15c.jpg)

Rest API wide
![Screen Shot 2022-07-14 at 3 46 05 PM](https://user-images.githubusercontent.com/1017872/178893791-64634ce5-2c21-4b2f-92df-982ff2525a75.jpg)

Rest API iPad
![Screen Shot 2022-07-14 at 3 45 47 PM](https://user-images.githubusercontent.com/1017872/178893771-4b3d949f-b059-484e-8754-2423566f9f7d.jpg)

adamwoodnz committed:
Revert removing Table of Contents from Code Reference and fix outside content instead

- Applies to ToC across all handbooks
- Fixed outside content on wide screens, otherwise inline below title

adamwoodnz committed:
Adjust ToC CSS to allow for sidebar and content overflow

tellyworth commented:
The breakpoint for the narrow ipad-style view seems to be with the main content area at about 900px, around 1500px for the window. That seems pretty wide for the ToC to jump over there, is that intended?

<img width="1998" alt="Screen Shot 2022-07-27 at 2 57 57 pm" src="https://user-images.githubusercontent.com/7200686/181165025-2d14f153-e811-4a56-b9cf-004d55f66db4.png">

adamwoodnz commented:
> The breakpoint for the narrow ipad-style view seems to be with the main content area at about 900px, around 1500px for the window. That seems pretty wide for the ToC to jump over there, is that intended?
> 
> <img alt="Screen Shot 2022-07-27 at 2 57 57 pm" width="1998" src="https://user-images.githubusercontent.com/7200686/181165025-2d14f153-e811-4a56-b9cf-004d55f66db4.png">

It is intended yes; that's the width that the ToC no longer fits alongside the content. We could look at an in-between layout where both the content and ToC widths are reduced so they can sit side by side down to around 1280px maybe.

Based on these [stats](https://gs.statcounter.com/screen-resolution-stats/desktop/worldwide) around 38% of people use a desktop screen > 1500px wide. Dropping the breakpoint to 1440 would net another 6% or 1366 would have a really significant impact:

![Screen Shot 2022-08-29 at 11 40 52 AM](https://user-images.githubusercontent.com/1017872/187099534-0317e626-c21b-4547-ba12-f348569e28ef.jpg)

StevenDufresne commented:
> It is intended yes; that's the width that the ToC no longer fits alongside the content. We could look at an in-between layout where both the content and ToC widths are reduced so they can sit side by side down to around 1280px maybe.

👍 I think you could even do it earlier, mimicking existing behavior at the "desktop" breakpoint which I think is `890px`.

adamwoodnz committed:
Reintroduce floating ToC in medium size screens
- High percentage of desktop users will see fixed ToC
- ToC width responds to viewport width

adamwoodnz commented:
> I think you could even do it earlier, mimicking existing behavior at the "desktop" breakpoint which I think is 890px

Yeah this width ends up working quite well for Code Reference pages. I've updated so we now have:

Below 890px -> full width inline ToC
890px to 1440px -> floating ToC (like existing)
1440px and above -> sticky ToC with responsive width

Chrome - Code Reference
![Kapture 2022-08-30 at 10 46 37](https://user-images.githubusercontent.com/1017872/187312348-1361abf4-57c8-4012-a782-ccdec69f15cc.gif)

Chrome - Coding Standards
![Kapture 2022-08-30 at 10 52 30](https://user-images.githubusercontent.com/1017872/187312803-d02aaad2-09b6-43f9-8247-d680245e8bd1.gif)

StevenDufresne commented on source/wp-content/themes/wporg-developer/scss/settings/_dimensions.scss:
```diff
@@ -0,0 +1,6 @@
+$single-handbook-content-width: 960px;
+$single-handbook-content-primary-width: 0.72;
```
Without looking at the implementation, this property is hard to understand.

StevenDufresne commented:
~~Viewing `coding-standards/inline-documentation-standards/javascript/`, at around `880px` the browser is clipping the Table of Contents on the right.~~

<img width="200" alt="Screen Shot 2022-08-30 at 3 39 05 PM" src="https://user-images.githubusercontent.com/1657336/187367328-0b88a69c-e36f-42b4-9425-951e2225635b.png">

**This was related to not having a sidebar configured. Since all the sites have a sidebar, I think we can disregard this for now.**

StevenDufresne commented:
The alignment is a bit distracting between the search bar, content and, ToC. I think it would be a nice improvement if we can get the content to align with the search bar. I'll take a look.

<img width="1225" alt="Screen Shot 2022-08-31 at 10 18 17 AM" src="https://user-images.githubusercontent.com/1657336/187572281-231d9eb8-ee34-4aca-8a99-e5d8e4d0a519.png">

StevenDufresne committed:
Harmonize site-branding and content padding to maintain a visual grid.

StevenDufresne commented:
I think this is good enough to merge. Since we are looking to release the codebase, I'll merge it so others can test it the project in its entirety. We can open up a PR for any additional issues.
